### PR TITLE
modify fedora template to start with secure boot support

### PR DESCRIPTION
**Pull request by Barakmor1**
**What this PR does / why we need it**:
modify fedora template that support UEFI BIOS with secure boot by default.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
- UEFI BIOS is work with fedora35+

**Release note**:

```release-note
modify fedora template to have secure boot support
```

### CODE REVIEW

**Comment by dominikholler** [comment]
@ksimon1 and @Barakmor1 Do you think we have to keep the fedora 34 template? Fedora 34 might be eol during may.

**Comment by Barakmor1** [comment]
> @ksimon1 and @Barakmor1 Do you think we have to keep the fedora 34 template? Fedora 34 might be eol during may.

Hey @dominikholler ,
I personally think we should keep fedora 34 for a while as fedora 35 is relatively new OS,
i think that this is out of the scope of this PR .

**Comment by ksimon1** [comment]
Currently fedora template supports versions 34 and 35. I would vote for updating fedora template to only use efi. @dominikholler  has valid point that F34 is going EOL in less than 3 months and 4.11 will be released after this date. Because otherways we will have to keep both templates with efi and without with different names (e.g. fedora-server-large-efi).

**Comment by Barakmor1** [comment]
> Currently fedora template supports versions 34 and 35. I would vote for updating fedora template to only use efi. @dominikholler has valid point that F34 is going EOL in less than 3 months and 4.11 will be released after this date. Because otherways we will have to keep both templates with efi and without with different names (e.g. fedora-server-large-efi).

Hey @ksimon1 ,
I tried to use the old template with efi as default but it doesn't work with fedora <= 34:
image

**Comment by Barakmor1** [comment]
> Currently fedora template supports versions 34 and 35. I would vote for updating fedora template to only use efi. @dominikholler has valid point that F34 is going EOL in less than 3 months and 4.11 will be released after this date. Because otherways we will have to keep both templates with efi and without with different names (e.g. fedora-server-large-efi).

Thanks!
I will update the PR soon .

**Comment by dominikholler** [comment]
@Barakmor1 does the template work with fedora 36, e.g. the qcow2 from https://download.fedoraproject.org/pub/fedora/linux/development/36/Cloud/x86_64/images/ ?

**Comment by Barakmor1** [comment]
> @Barakmor1 does the template work with fedora 36, e.g. the qcow2 from https://download.fedoraproject.org/pub/fedora/linux/development/36/Cloud/x86_64/images/ ?

i will try it and update :)

**Comment by dominikholler** [comment]
@Barakmor1 I wonder if this change applies only to fedora, or if other linux distributions, e.g. centos stream 9, are affected, too.
Do you think it is implementable to derive the decision if efi or bios should be used from libosinfo?

**Comment by dominikholler** [comment]
> Do you think it is implementable to derive the decision if efi or bios should be used from libosinfo?

No, the libosinfo attribute seems to refer to the guest os in general, not to the cloud image.

**Comment by Barakmor1** [comment]
@dominikholler 
Hey,
I updated the PR to modify the old template instead of adding new one,
also i tested it and the new template work with fedora 36.

File: generate-templates.yaml [hunk 1]
```diff
@@ -238,7 +238,7 @@
 
   - name: Load Fedora versions
     set_fact:
-      fedora_labels: "{{ lookup('osinfo', 'distro=fedora') |select('osinfo_active') |map(attribute='short_id') |select('match', '^fedora\\d+$') |list |sort }}"
+      fedora_labels: "{{ lookup('osinfo', 'distro=fedora') |select('osinfo_active') |map(attribute='short_id') |select('match', '^fedora3[5-9]|^fedora[4-9][0-9]') |list |sort }}"
```
Comment (dominikholler) [inline generate-templates.yaml:242]: Probably this can be simplified again after the release of fedora36

**Review by dominikholler — APPROVED** [review]
/lgtm

**Pull request merged**